docs/engine.md

```diff
@@ -19,14 +32,29 @@ Currently, the engines that are supported by DJL are:
 - [PaddlePaddle](../engines/paddlepaddle/README.md) - supports basic inference
 - [TFLite](../engines/tflite/tflite-engine/README.md) - supports basic inference
 - [TensorRT](../engines/tensorrt/README.md) - supports basic inference
+- [XGBoost](../engines/ml/xgboost/README.md) - supports basic inference
+- [LightGBM](../engines/ml/lightgbm/README.md) - supports basic inference
 - [DLR](../engines/dlr/README.md) - supports basic inference
+- [Python](https://github.com/deepjavalibrary/djl-serving/blob/master/engines/python/README.md) - supports basic inference
 
 ## Setup
 
-In order to choose an engine, it must be added into the Java classpath. Usually this means additional Maven or Gradle dependencies. Many engines require multiple dependencies be added, so look at the engine README for your desired engine to learn what dependencies are necessary.
-
-It is also possible to load multiple engines simultaneously. When DJL starts up, it chooses a default engine from the available engines. Most of the API that requires an engine such as `NDManager.newBaseManager()` or `Model.newInstance()` will internally use the default engine. As of DJL 10.0, it chooses the default engine based on a ranking of how much we recommend the engine, but before that it was chosen at random. For those calls, you can also choose the engine manually by getting an engine with `Engine.getEngine(engineName)` and then calling the equivalent method on the Engine class such as `engine.newBaseManager()`.
-
-Some calls will also take advantage of all possible engines. For example, model loading will try all of the engines available to see if any work for the model you are trying to load.
-
-You can also choose the default engine manually. Each engine has a name which can be found in the engine's javadoc or README. You can set the default engine by setting either the "DJL_DEFAULT_ENGINE" environment variable or the "ai.djl.default_engine" Java property. Either one should be set to the name of the desired default engine.
+In order to choose an engine, it must be added into the Java classpath. Usually this means
+additional Maven or Gradle dependencies. Many engines require multiple dependencies be added,
+so look at the engine README for your desired engine to learn what dependencies are necessary.
+
+It is also possible to load multiple engines simultaneously. When DJL starts up, it chooses a
+default engine from the available engines. Most of the API that requires an engine such as
+`NDManager.newBaseManager()` or `Model.newInstance()` will internally use the default engine.
+DJL chooses the default engine based on a ranking of how much we recommend the engine, but before
+that it was chosen at random. For those calls, you can also choose the engine manually by getting
```

Comment:
Should we include when (i.e. version number) this behavior changed? Or maybe we omit the detail about how it used to work and just state that we choose based on rank if no override is passed in?

Reply:
0.10.0 has been released for quite some time. Since we move fast, we'd like to document the state of the latest version and ignore the old behaviors.